src/Maker/MakeDto.php

```diff
@@ -0,0 +1,250 @@
+            'Data'
+        );
+
+        $boundClass = $input->getArgument('bound-class');
+
+        $boundClassDetails = $generator->createClassNameDetails(
+            $boundClass,
+            'Entity\\'
+        );
+
+        // get some doctrine details
+        $doctrineEntityDetails = $this->entityHelper->createDoctrineDetails($boundClassDetails->getFullName());
+
+        if (null === $doctrineEntityDetails) {
+            $io->error([
+                'The bound class is not a valid doctrine entity.',
+            ]);
+
+            return;
+        }
+
+        // get class metadata (used by regenerate)
+        $metaData = $this->entityHelper->getMetadata($boundClassDetails->getFullName());
+
+        // list of fields
+        $fields = $metaData->fieldMappings;
+
+        $boundClassVars = [
+            'bounded_full_class_name' => $boundClassDetails->getFullName(),
+            'bounded_class_name' => $boundClassDetails->getShortName(),
+        ];
+
+        // the result is passed to the template
+        $addHelpers = $io->confirm('Add helper extract/fill methods?');
+
+        // Generate getters/setters
+        $omitGettersSetters = $io->confirm('Omit generation of getters/setters?');
+
+        // filter id from fields
+        $fields = array_filter($fields, function ($field) {
+            // mapping includes id field when property is an id
+            if (!empty($field['id'])) {
+                return false;
+            }
+
+            return true;
+        });
+
+        $DTOClassPath = $generator->generateClass(
+            $dataClassNameDetails->getFullName(),
+            __DIR__.'/../Resources/skeleton/dto/Data.tpl.php',
+            array_merge(
+                [
+                    'fields' => $fields,
+                    'addHelpers' => $addHelpers,
+                    'omitGettersSetters' => $omitGettersSetters,
+                ],
+                $boundClassVars
+            )
+        );
+
+        $generator->writeChanges();
+        $manipulator = $this->createClassManipulator($DTOClassPath, $omitGettersSetters);
+        $mappedFields = $this->getMappedFieldsInEntity($metaData);
+
+        // Did we import assert annotations?
+        $assertionsImported = false;
+
+        foreach ($fields as $fieldName => $mapping) {
+            if (!\in_array($fieldName, $mappedFields)) {
+                continue;
+            }
+
+            $annotationReader = new AnnotationReader();
+
+            // Lookup classname for inherited properties
+            if (array_key_exists('declared', $mapping)) {
+                $fullClassName = $mapping['declared'];
+            } else {
+                $fullClassName = $boundClassDetails->getFullName();
+            }
+
+            // Property Annotations
+            $reflectionProperty = new \ReflectionProperty($fullClassName, $fieldName);
+            $propertyAnnotations = $annotationReader->getPropertyAnnotations($reflectionProperty);
```

Comment:
So, there are a few issues with the copying of the validation constraints:

1) Validation metadata may be stored as XML or YML, and we're not covering that - or even warning the user that we don't handle that situation.
2) I believe you can use the `validator.mapping.class_metadata_factory` service (which gives you this method - https://github.com/symfony/symfony/blob/1fc577f2c7c1cff2857658537adf4f30a3ad60ac/src/Symfony/Component/Validator/Validator/RecursiveValidator.php#L81 ) to read the metadata. This will include XML and YAML metadata, but I don't know if we will know the source of the metadata (though, we could do a simple check on the class for one of the "found" constraints to see if its there - to know if annotations are used).

We probably should not try to add XML/YAML validation metadata, if that's what the user is currently using. However, we COULD do annotations (like you are now). But, if we do that, we also need to show a big warning to the user if we detect validation metadata that is XML or YAML - i.e. "The validation constraints were not copied to the new class" (not perfectly explained, but hopefully you get the idea).